homeassistant/components/bayesian/config_flow.py

```diff
@@ -0,0 +1,208 @@
+
+
+async def _validate_mode(
+    handler: SchemaCommonFlowHandler, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the threshold mode, and set limits to None if not set."""
+    return {**user_input}
+
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_PROBABILITY_THRESHOLD, default=DEFAULT_PROBABILITY_THRESHOLD * 100
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Required(
+            CONF_PRIOR, default=DEFAULT_PROBABILITY_THRESHOLD * 100
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Optional(CONF_DEVICE_CLASS): selector.SelectSelector(
+            selector.SelectSelectorConfig(
+                options=[cls.value for cls in BinarySensorDeviceClass],
+                mode=selector.SelectSelectorMode.DROPDOWN,
+                translation_key="binary_sensor_device_class",
+                sort=True,
+            ),
+        ),
+    }
+)
+SUBSCHEMA_BOILERPLATE = vol.Schema(
+    {
+        vol.Required(CONF_P_GIVEN_T): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+        vol.Required(CONF_P_GIVEN_F): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.SLIDER,
+                step=1.0,
+                min=0,
+                max=100,
+                unit_of_measurement="%",
+            ),
+        ),
+    }
+)
+
+NUMERIC_STATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY_ID): selector.EntitySelector(
+            selector.EntitySelectorConfig(
+                domain=[SENSOR_DOMAIN, INPUT_NUMBER_DOMAIN, NUMBER_DOMAIN]
+            )
+        ),
+        vol.Optional(CONF_ABOVE): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.BOX, step="any"
+            ),
+        ),
+        vol.Optional(CONF_BELOW): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                mode=selector.NumberSelectorMode.BOX, step="any"
+            ),
+        ),
+    },
+).extend(SUBSCHEMA_BOILERPLATE.schema)
+
+STATE_SUBSCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ENTITY_ID): selector.EntitySelector(
+            selector.EntitySelectorConfig(domain=[SENSOR_DOMAIN, BINARY_SENSOR_DOMAIN])
```

Comment:
Are these the only sensor domains we want to support?

Reply:
In [5b0d09f](https://github.com/home-assistant/core/pull/122552/commits/5b0d09f9fc5570fb5cda97478bb13204ed75bef5) I have enabled many more domains in the absence of a centrally managed list of all possible domains. Broadly I have enabled all domains for `state` observations that a user could reasonably predict e.g those that are not timestamps
For numeric domains I have included all those which have a state that is usually a number.